client/src/App.vue

```diff
@@ -25,7 +26,8 @@ export default defineComponent({
   setup() {
     // Create and provide default state. This is updated once API data is fetched.
     const initialDataLayer: DataLayer = null as unknown as DataLayer;
-    const state = reactive(new State([], initialDataLayer));
+    const initialMap: mapboxgl.Map = null as unknown as mapboxgl.Map;
```

Comment:
What does the `as unknown` do here? It makes me feel like there's some weirdness going on with types. Maybe the `map` property of `State` should be nullable so this isn't necessary.

Reply:
Yeah on a previous PR Justine left a comment about just putting null as Type instead of making the type like 'type | null' to avoid having to account for null values everywhere the field is accessed. I'll switch this to use Map | null so it is just nullable.

Reply:
I believe making it nullable is the "right" way of doing things. Right now it's effectively nullable, but without any of the safeguards that making it explicitly nullable would provide.

This applies to each of the below cases too. It should either be nullable or have an empty, but valid instance of that type.

Reply:
SG will update others as well.